std/file.d

```diff
@@ -1072,13 +1073,18 @@ SysTime timeLastModified(in char[] name) @safe
     }
     else version(Posix)
     {
-        static auto trustedStat(in char[] path, ref stat_t buf) @trusted
+        auto namez = name.tempCString!FSChar();
+        static auto trustedStat(const(FSChar)* namez, ref stat_t buf) @trusted
         {
-            return stat(path.tempCString(), &buf);
+            return stat(namez, &buf);
         }
         stat_t statbuf = void;
 
-        cenforce(trustedStat(name, statbuf) == 0, name);
+        static if (isNarrowString!R && is(Unqual!(ElementEncodingType!R) == char))
```

Comment:
How about at least making it a simpler private helper:
```
enum isUtf8String(R) = is(R : char[]) || is(R : const(char)[]) || is(R: immutable(char)[]);
```
It's going to have a lot of use internally in this module.